### PR TITLE
Adjustment for last partition bug

### PR DESCRIPTION
**Pull request by knighton**

### CODE REVIEW

File: streaming/base/index.py [hunk 1]
```diff
@@ -205,6 +205,14 @@ def _get_min_max_size(start: int, total: int, part: int, num_parts: int):
             worker_max_id = \
                 device_min_id + (worker_max_batch_id + 1) * self.batch_size - max_id_offset - 1
 
+            # Adjustment for last partition.
+            if self.total_samples == worker_max_id:
+                if 0 < worker_min_id:
```
Comment (karan6181) [inline streaming/base/index.py:210]: Just my preference: sometimes hard to follow `0 < worker_min_id`. Maybe `worker_min_id > 0` is easier to read ?

File: streaming/base/index.py [hunk 2]
```diff
@@ -205,6 +205,14 @@ def _get_min_max_size(start: int, total: int, part: int, num_parts: int):
...
+                    worker_min_id -= 1
+                worker_max_id -= 1
+            elif self.total_samples < worker_max_id:
+                raise RuntimeError('Invalid partitioning')
```
Comment (karan6181) [inline streaming/base/index.py:214]: Is RuntimeError the right exception or ValueError or any other exception?

File: streaming/base/index.py (hunk 1 above)
Comment (knighton) [reply streaming/base/index.py:210]: Done

File: streaming/base/index.py (hunk 2 above)
Comment (knighton) [reply streaming/base/index.py:214]: Done

**Review by karan6181 — APPROVED** [review]
LGTM! Thanks!
